vending-app/src/main/kotlin/com/google/android/finsky/DeviceSyncInfo.kt

```diff
@@ -0,0 +1,584 @@
+                builder.reqTouchScreen(configurationInfo.reqTouchScreen)
+            }
+            if (configurationInfo.reqKeyboardType != Configuration.KEYBOARD_UNDEFINED) {
+                builder.reqKeyboardType(configurationInfo.reqKeyboardType)
+            }
+            if (configurationInfo.reqNavigation != Configuration.NAVIGATION_UNDEFINED) {
+                builder.reqNavigation(configurationInfo.reqNavigation)
+            }
+            builder.reqGlEsVersion(configurationInfo.reqGlEsVersion)
+            builder.reqInputFeaturesV1((configurationInfo.reqInputFeatures and 1) == 1)
+                .reqInputFeaturesV2((configurationInfo.reqInputFeatures and 2) > 0)
+        }
+        val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+        val size = Point()
+        windowManager.defaultDisplay.getSize(size)
+        builder.displayX(size.x).displayY(size.y)
+        if (SDK_INT >= 24) {
+            builder.deviceStable(DisplayMetrics.DENSITY_DEVICE_STABLE)
+                .deviceStablePoint(calculatePoint(size, DisplayMetrics.DENSITY_DEVICE_STABLE))
+        }
+        val configuration = context.resources.configuration
+        builder.screenLayout(configuration.screenLayout)
+            .smallestScreenWidthDp(configuration.smallestScreenWidthDp)
+            .systemSharedLibraryNames(listOf(*Objects.requireNonNull(context.packageManager.systemSharedLibraryNames)))
+            .locales(listOf(*context.assets.locales))
+        if (SDK_INT >= 24) {
+            builder.glExtensions(gpuInfoList.stream()
+                .flatMap { fetchedGlStrings: FetchedGlStrings -> fetchedGlStrings.glExtensions?.let { Arrays.stream(it.toTypedArray()) } }
+                .collect(Collectors.toList()))
+                .isLowRamDevice(activityManager.isLowRamDevice)
+        }
+        val memoryInfo = ActivityManager.MemoryInfo()
+        activityManager.getMemoryInfo(memoryInfo)
+        builder.totalMem(memoryInfo.totalMem)
+            .availableProcessors(Runtime.getRuntime().availableProcessors())
+        val systemAvailableFeatures = context.packageManager.systemAvailableFeatures
+        for (featureInfo in systemAvailableFeatures) {
+            if (!TextUtils.isEmpty(featureInfo.name)) {
+                var featureInfoProto = FeatureInfoProto.Builder().build()
+                if (SDK_INT >= 24) {
+                    featureInfoProto = FeatureInfoProto.Builder().name(featureInfo.name).version(featureInfo.version).build()
+                }
+                builder.featureInfoList = builder.featureInfoList.toMutableList().apply {
+                    add(featureInfoProto)
+                }
+                builder.featureNames = builder.featureNames.toMutableList().apply {
+                    add(featureInfoProto.name!!)
+                }
+            }
+        }
+        if (SDK_INT >= 21) {
+            builder.supportedAbi(listOf(*Build.SUPPORTED_ABIS))
+        }
```

Comment:
@mar-v-in
Since this may be used to download things (and we don't want to download things incompatible with the device) in case we simulate a device >= 21 from a device < 21 we should generate a SUPPORTED_ABIS list based on the real device instead of the spoofed one.
What do you think?

Reply:
If the profile diverts from the actual device too much, this would indeed likely cause issues, but that is to be expected elsewhere as well. I think we can keep it as is and revisit later if and how we want to adjust this device sync feature - as likely we want to extend the device profiles to cover more of the elements that we send here.